internal/controller/watcher/controller.go

```diff
@@ -122,6 +122,8 @@ func (r *Reconciler) stateHandling(ctx context.Context, watcher *v1beta2.Watcher
 		return r.handleProcessingState(ctx, watcher)
 	case shared.StateReady, shared.StateWarning:
 		return r.handleProcessingState(ctx, watcher)
+	case shared.StateUnmanaged:
+		panic("unmanaged not a valid state for watcher")
```

Comment:
same reason as [here](https://github.com/kyma-project/lifecycle-manager/pull/1717/files#r1735776926), btw, Kyma and Watcher CR are all controlled by us, no one will change the state to unmanaged manually, fallthrough this condition and stop requeue is enough already.

Reply:
Yes, I wanted to indicate that this state should never happen and if it happens it's a developer error and not an error we should handle. Rather bubble it up as quickly as possible.

Reply:
But one of the key aspect of error handling is reporting, this also aplies to fail fast principle, it expects that failures are detected early and reported immediately.  The current use of panic only stops the process from continuing but does not properly handle the reporting.

If the Kyma CR is in this state, using panic will not cause the KLM to crash (which is also a poor design choice in my opinion), but will only stop the current worker. This means that without checking the logs, it is difficult to identify that such an error exists.

As I mentioned [here](https://github.com/kyma-project/lifecycle-manager/pull/1717#discussion_r1735776926), a proper handling approach would be to write an error message in the Kyma CR and **prevent it from requeuing**. This would avoid interfering with other Kyma CRs and make the KLM more robust.

Reply:
Alright! Will change to returning an error instead 👍